content/guides/collaborations/connect-slack-to-group-collabs/4-handle-slack-events.md

```diff
@@ -0,0 +1,426 @@
+    });
+    res.send('Adding content');
+  } else {
+    res.send('Invalid input. Example usage: /boxadd file 123456');
+  }
+} else {
+  res.send('Invalid action');
+}
+```
+<!-- markdownlint-enable line-length -->
+
+The purpose of this function is to figure out if the payload from Slack is a
+user event or a Slash command, fetch any needed information, then route to the
+appropriate function to process the results.
+
+If the payload is a user event, denoted by `data.type` being set to
+`event_callback`, we extract a few pieces of information.
+
+* `eventType`: The type of event to determine if a user is leaving
+ (`member_left_channel`) or joining (`member_joined_channel`) the channel. 
+* `channel`: The channel ID, which will be used as the Box group name.
+* `userId`: The ID of the user, to look up their profile email which will bind
+ to a user profile in Box that uses the same email.
+
+The process function then fetches the profile of the user by calling
+`getSlackUser`, and once obtained that user profile is sent to the
+`processUser` function to add or remove them from the Box group.
+
+If the payload is a slash command, denoted by `data.command` being set to
+`/boxadd`, the content of the command that represents the Box ID and whether
+it's a file or folder, such as `file 1234`, is extracted and split to get the
+individual values. Those values are validated for proper content.
+
+Once validated, the profile of the Slack user is obtained to get the email,
+then the user profile is sent to `processContent` to collaborate the Box
+content in with the Box group so that everyone has access. 
+
+<Message type='notice'>
+  The reason we want to obtain the Slack email here is that when we're sharing
+  content to the group we want users to be able to share files and folders from
+  their own accounts, not the application service account. We need to match the
+  Slack email against the Box user account email to do that.
+</Message>
+
+</Choice>
+<Choice option='programming.platform' value='java' color='none'>
+
+```java
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='dotnet' color='none'>
+
+```dotnet
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='python' color='none'>
+
+```python
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='ruby' color='none'>
+
+```ruby
+
+```
+
+</Choice>
+<Choice option='programming.platform' unset color='none'>
+  <Message danger>
+    # Incomplete previous step
+    Please select a preferred language / framework in step 1 to get started.
+  </Message>
+</Choice>
+
+## Process Slack channel users
+
+<Choice option='programming.platform' value='node' color='none'>
+
+When a bot is first added to a channel, it needs to run an audit of all users
+currently in the channel and create a Box group with those people in order to
+create a baseline for the channel.
+
+In the `processSlackChannel` function, replace the
```

Comment:
I don't see where this code is called from.

Reply:
Added missing function to scaffolding